addons/binding/org.openhab.binding.jeelink/src/main/java/org/openhab/binding/jeelink/handler/JeeLinkSensorHandler.java

```diff
@@ -0,0 +1,231 @@
+        String idStr = getConfig().as(JeeLinkSensorConfig.class).sensorId;
+        try {
+            updateStatus(ThingStatus.OFFLINE);
+
+            Average<R> avg = null;
+            if (bufferSize > 0) {
+                logger.debug("Using rolling average with buffer size {}...", bufferSize);
+                avg = new RollingReadingAverage<>(clazz, bufferSize);
+            }
+
+            logger.debug("Adding reading listener for id {}...", idStr);
+            listener = new JeeLinkReadingListener<R>(avg) {
+                @Override
+                public synchronized void handleReading(R reading) {
+                    try {
+                        if (isReadingWithinBounds(reading)) {
+                            boolean initial = lastReading == null;
+
+                            // propagate initial reading
+                            if (initial) {
+                                updateReadingStates(reading);
+                            }
+                            super.handleReading(reading);
+
+                            // propagate every reading in live mode
+                            if (!initial && updateInterval == 0) {
+                                updateReadingStates(getCurrentReading());
+                            }
+                        }
+
+                        // make sure status is online as soon as we get a reading
+                        updateStatus(ThingStatus.ONLINE);
+                    } catch (Throwable th) {
+                        logger.error("Uncaught throwable in JeeLink sensor handler for thing " + getThing().getLabel()
+                                + " (" + getThing().getUID() + ")...", th);
+                    }
+                }
+            };
+
+            JeeLinkHandler jlh = (JeeLinkHandler) getBridge().getHandler();
+            JeeLinkReadingConverter<R> c = jlh.getConverter(clazz);
+            c.addReadingListener(idStr, listener);
+            initialized = true;
+        } catch (NumberFormatException ex) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Thing has an invalid sensor id: " + idStr);
+            logger.info("JeeLink sensor handler for thing {} ({}) is OFFLINE", getThing().getLabel(),
+                    getThing().getUID());
+        }
+    }
+
+    private ScheduledFuture<?> createUpdateJob(final int updateInterval) {
+        Runnable runnable = new Runnable() {
+            @Override
+            public void run() {
+                logger.trace("JeeLink sensor cyclic propagation for thing {} ({})...", getThing().getLabel(),
+                        getThing().getUID());
+
+                if (listener != null) {
+                    try {
+                        updateReadingStates(listener.getCurrentReading());
+                    } catch (Throwable th) {
+                        logger.error("Uncaught throwable in JeeLink sensor handler for thing " + getThing().getLabel()
```

Comment:
Self composed logging string, can you check your branch with the check report which has been integrated in the build, after a `mvn clean install` your `target` directory should contain a `code-analysis/report.html`